packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -177,6 +175,20 @@ export const useShippingMethod = props => {
         }
     }, [chosenShippingMethodData]);
 
+    // Simple heuristic to check shipping data existed prior to this render.
+    // On first submission, when we have data, we should tell the checkout page
+    // so that we set the next step correctly.
+    const hasData =
+        !!chosenShippingMethodData &&
+        !!chosenShippingMethodData.cart.shipping_addresses[0]
+            .selected_shipping_method;
```

Comment:
Our components together guard against this use case, but someone just using this talon might still enter this block w/o setting an address. Let's guard against it (don't apply this from GitHub, I'm sure formatting is off).
```suggestion
        !!chosenShippingMethodData.cart.shipping_addresses.length && chosenShippingMethodData.cart.shipping_addresses[0].selected_shipping_method;
```

Reply:
Good call. I'll make the change.

Reply:
Fixing in 6130923b2